asgi_rabbitmq/core.py

```diff
@@ -957,16 +977,17 @@ def _schedule(self, f, *args, **kwargs):
         def receive_twisted(self, channels):
             """Twisted-native implementation of receive."""
 
-            deferred = defer.Deferred()
-
-            def resolve_deferred(future):
-
-                reactor.callFromThread(deferred.callback, future.result())
-
-            future = self.thread.schedule(RECEIVE, channels, block=True)
-            future.add_done_callback(resolve_deferred)
-            defer.returnValue((yield deferred))
-
+            while True:
+                try:
+                    result = yield deferred_from_future(
+                        self.thread.schedule(RECEIVE, channels, block=True)
+                    )
+                    defer.returnValue(result)
+                except ConnectionClosed:
+                    delay = self.thread.connection.parameters.retry_delay
+                    logger.warning(
+                        'Disconnected, retrying after %s seconds', delay)
+                    deferred_sleep(delay)
```

Comment:
Shouldn't it be used with `yield` to work with deferred & `inlineCallbacks` properly?

Reply:
yes, my mistake

Reply:
updated